jupyterhub/spawners.py

```diff
@@ -270,8 +270,10 @@ def _expand_user_properties(self, template):
         < 63 character pod names. We keep the last 16 characters which is
         the server name hash.
 
-        Code adapted from
-        https://github.com/jupyterhub/kubespawner/blob/master/kubespawner/spawner.py
+        If the string being processed is not the pod or pvc template,
+        let it pass through.
         """
-        rendered = super()._expand_user_properties(template)
-        return rendered if len(rendered) <= 63 else rendered[:47] + rendered[-16:]
+        if template == self.pod_name_template or template == self.pvc_name_template:
```

Comment:
Shouldn't we call the base class method anyway and only doing the shortening if this condition holds:
```
rendered = super()._expand_user_properties(template)
if template == self.pod_name_template or template == self.pvc_name_template:
    return rendered if len(rendered) <= 63 else rendered[:47] + rendered[-16:]
return rendered
```

Reply:
hmm maybe - but annotations, for example, don't need to be escaped at all and the jupyterhub code is passing them through this function. That means that you might get a different annotation than you expect (I will make an issue/PR to kubespawner about this). What's weird is that they pass the annotation values through `expand_user_properties` but it's the keys in annotations that have character and length limitations, not the values.

Reply:
On second thought I think you're right, since the rest of the code expects that behavior we should leave it for now. I will change it.